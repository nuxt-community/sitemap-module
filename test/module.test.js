const { readFileSync } = require('fs')
const { resolve } = require('path')
const { gunzipSync } = require('zlib')

const { Nuxt, Builder, Generator } = require('nuxt')
const request = require('request-promise-native')

const config = require('./fixture/nuxt.config')
config.dev = false
config.modules = [require('..')]
config.sitemap = {}

const url = (path) => `http://localhost:3000${path}`
const get = (path, options = null) => request(url(path), options)
const getGzip = (path) => request({ url: url(path), encoding: null })

const startServer = async (config) => {
  const nuxt = new Nuxt(config)
  await nuxt.ready()
  await new Builder(nuxt).build()
  await nuxt.listen(3000)
  return nuxt
}
const runGenerate = async (config) => {
  const nuxt = new Nuxt(config)
  await nuxt.ready()
  const builder = new Builder(nuxt)
  const generator = new Generator(nuxt, builder)
  await generator.generate()
}

jest.setTimeout(60000)

describe('ssr - pages', () => {
  test('should render all pages', async () => {
    const nuxt = await startServer(config)

    // static routes
    let html = await get('/')
    expect(html).toContain('/index')
    html = await get('/sub/')
    expect(html).toContain('/sub/index')
    html = await get('/sub/sub')
    expect(html).toContain('/sub/sub')

    // static child-routes
    html = await get('/parent')
    expect(html).toContain('/parent')
    html = await get('/parent/child')
    expect(html).toContain('/parent/child')
    html = await get('/parent/child/subchild')
    expect(html).toContain('/parent/child/subchild')

    // dynamic routes
    html = await get('/child/')
    expect(html).toContain('/child/index')
    html = await get('/child/1')
    expect(html).toContain('/child/1')
    html = await get('/1/')
    expect(html).toContain('/1/index')

    // excluded routes
    html = await get('/exclude')
    expect(html).toContain('/exclude')

    // filtered routes
    html = await get('/filtered')
    expect(html).toContain('/filtered')

    await nuxt.close()
  })
})

describe('sitemap - init options', () => {
  let nuxt = null
  let xml = null

  afterEach(async () => {
    await nuxt.close()
  })

  test('as object', async () => {
    nuxt = await startServer({
      ...config,
      sitemap: {
        hostname: 'https://example.com/',
      },
    })

    xml = await get('/sitemap.xml')
    expect(xml).toContain('<loc>https://example.com/</loc>')
  })

  test('as array', async () => {
    nuxt = await startServer({
      ...config,
      sitemap: [
        {
          hostname: 'https://example.com/',
        },
      ],
    })

    xml = await get('/sitemap.xml')
    expect(xml).toContain('<loc>https://example.com/</loc>')
  })

  test('as function', async () => {
    nuxt = await startServer({
      ...config,
      sitemap: () => ({
        hostname: 'https://example.com/',
      }),
    })

    xml = await get('/sitemap.xml')
    expect(xml).toContain('<loc>https://example.com/</loc>')
  })

  test('as boolean', async () => {
    nuxt = await startServer({
      ...config,
      sitemap: false,
    })

    xml = await get('/sitemap.xml')
    expect(xml).toContain('<!doctype html>')
    expect(xml).not.toContain('<loc>https://example.com/</loc>')
  })

  test('as boolean from function', async () => {
    nuxt = await startServer({
      ...config,
      sitemap: () => false,
    })

    xml = await get('/sitemap.xml')
    expect(xml).toContain('<!doctype html>')
    expect(xml).not.toContain('<loc>https://example.com/</loc>')
  })
})

describe('sitemap - minimal configuration', () => {
  test('sitemap.xml', async () => {
    const nuxt = await startServer({
      ...config,
      generate: {
        routes: null,
      },
      sitemap: {
        hostname: 'https://example.com/',
      },
    })

    const xml = await get('/sitemap.xml')
    expect(xml).toMatchSnapshot()

    await nuxt.close()
  })
})

describe('sitemap - advanced configuration', () => {
  let nuxt = null
  let xml = null

  afterEach(async () => {
    await nuxt.close()
  })

  describe('custom options', () => {
    beforeAll(async () => {
      nuxt = await startServer({
        ...config,
        sitemap: {
          path: '/custom-sitemap.xml',
          hostname: 'https://example.com/',
          exclude: ['/exclude'],
          routes: ['1/', 'child/1', { url: 'test/' }, { route: '/payload/1', payload: { id: 1 } }],
          filter: ({ routes }) => routes.filter((route) => route.url !== '/filtered'),
          defaults: {
            changefreq: 'daily',
            priority: 1,
          },
          xmlNs: 'xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"',
          xslUrl: 'sitemap.xsl',
          gzip: false,
          cacheTime: 0,
        },
      })
    })

    test('custom path', async () => {
      xml = await get('/custom-sitemap.xml')
      expect(xml).toContain('<loc>https://example.com/</loc>')
    })

    test('static routes', () => {
      // static routes
      expect(xml).toContain('<loc>https://example.com/</loc>')
      expect(xml).toContain('<loc>https://example.com/sub</loc>')
      expect(xml).toContain('<loc>https://example.com/sub/sub</loc>')

      // static child-routes
      expect(xml).toContain('<loc>https://example.com/parent</loc>')
      expect(xml).toContain('<loc>https://example.com/parent/child</loc>')
      expect(xml).toContain('<loc>https://example.com/parent/child/subchild</loc>')
      expect(xml).not.toContain('<loc>https://example.com/parent/</loc>')
      expect(xml).not.toContain('<loc>https://example.com/parent/child/</loc>')
    })

    test('dynamic routes', () => {
      expect(xml).toContain('<loc>https://example.com/child</loc>')
      expect(xml).toContain('<loc>https://example.com/child/1</loc>')
      expect(xml).toContain('<loc>https://example.com/1/</loc>')
      expect(xml).toContain('<loc>https://example.com/test/</loc>')
    })

    test('excluded routes', () => {
      expect(xml).not.toContain('<loc>https://example.com/exclude</loc>')
    })

    test('filtered routes', () => {
      expect(xml).not.toContain('<loc>https://example.com/filtered</loc>')
    })

    test('default options', () => {
      expect(xml).toContain(
        '<url><loc>https://example.com/</loc><changefreq>daily</changefreq><priority>1.0</priority></url>'
      )
      expect(xml).toContain(
        '<url><loc>https://example.com/child</loc><changefreq>daily</changefreq><priority>1.0</priority></url>'
      )
    })

    test('custom XML namespaces', () => {
      expect(xml).toContain('<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">')
    })

    test('custom XSL', () => {
      expect(xml).toContain('<?xml-stylesheet type="text/xsl" href="sitemap.xsl"?>')
    })
  })

  describe('custom options', () => {
    test('etag enabled', async () => {
      nuxt = await startServer({
        ...config,
        sitemap: {
          gzip: true,
        },
      })

      const requestOptions = {
        simple: false,
        resolveWithFullResponse: true,
      }

      // 1st call
      let response = await get('/sitemap.xml', requestOptions)
      expect(response.statusCode).toEqual(200)
      expect(response.headers.etag).not.toBeUndefined()
      // 2nd call
      response = await get('/sitemap.xml', {
        headers: {
          'If-None-Match': response.headers.etag,
        },
        ...requestOptions,
      })
      expect(response.statusCode).toEqual(304)

      // 1st call
      response = await get('/sitemap.xml.gz', requestOptions)
      expect(response.statusCode).toEqual(200)
      expect(response.headers.etag).not.toBeUndefined()
      // 2nd call
      response = await get('/sitemap.xml.gz', {
        headers: {
          'If-None-Match': response.headers.etag,
        },
        ...requestOptions,
      })
      expect(response.statusCode).toEqual(304)
    })

    test('etag disabled', async () => {
      nuxt = await startServer({
        ...config,
        sitemap: {
          etag: false,
          gzip: true,
        },
      })

      const requestOptions = {
        simple: false,
        resolveWithFullResponse: true,
      }

      let response = await get('/sitemap.xml', requestOptions)
      expect(response.statusCode).toEqual(200)
      expect(response.headers.etag).toBeUndefined()

      response = await get('/sitemap.xml.gz', requestOptions)
      expect(response.statusCode).toEqual(200)
      expect(response.headers.etag).toBeUndefined()
    })

    test('gzip enabled', async () => {
      nuxt = await startServer({
        ...config,
        sitemap: {
          gzip: true,
        },
      })

      const xml = await get('/sitemap.xml')
      const gz = await getGzip('/sitemap.xml.gz')
      const sitemap = gunzipSync(gz).toString()
      expect(xml).toEqual(sitemap)
    })

    test('trailingSlash enabled', async () => {
      nuxt = await startServer({
        ...config,
        sitemap: {
          hostname: 'https://example.com',
          trailingSlash: true,
          routes: ['test'],
        },
      })

      const xml = await get('/sitemap.xml')
      expect(xml).not.toContain('<loc>https://example.com/sub</loc>')
      expect(xml).not.toContain('<loc>https://example.com/sub/sub</loc>')
      expect(xml).not.toContain('<loc>https://example.com/test</loc>')
      expect(xml).toContain('<loc>https://example.com/sub/</loc>')
      expect(xml).toContain('<loc>https://example.com/sub/sub/</loc>')
      expect(xml).toContain('<loc>https://example.com/test/</loc>')
    })
  })

  describe('i18n options', () => {
    const modules = [require('nuxt-i18n'), require('..')]

    const nuxtI18nConfig = {
      locales: ['en', 'fr'],
      defaultLocale: 'en',
    }

    const sitemapConfig = {
      hostname: 'https://example.com',
      trailingSlash: true,
      i18n: 'en',
      routes: ['foo', { url: 'bar' }],
    }

    test('strategy "no_prefix"', async () => {
      nuxt = await startServer({
        ...config,
        modules,
        i18n: {
          ...nuxtI18nConfig,
          strategy: 'no_prefix',
        },
        sitemap: sitemapConfig,
      })

      const xml = await get('/sitemap.xml')
      expect(xml).toContain('<loc>https://example.com/</loc>')
      expect(xml).not.toContain('<loc>https://example.com/en/</loc>')
      expect(xml).not.toContain('<loc>https://example.com/fr/</loc>')
      expect(xml).not.toContain('<xhtml:link rel="alternate" hreflang="en" href="https://example.com/"/>')
      expect(xml).not.toContain('<xhtml:link rel="alternate" hreflang="en" href="https://example.com/en/"/>')
      expect(xml).not.toContain('<xhtml:link rel="alternate" hreflang="fr" href="https://example.com/fr/"/>')
      expect(xml).not.toContain('<xhtml:link rel="alternate" hreflang="x-default" href="https://example.com/"/>')
    })

    test('strategy "prefix"', async () => {
      nuxt = await startServer({
        ...config,
        modules,
        i18n: {
          ...nuxtI18nConfig,
          strategy: 'prefix',
        },
        sitemap: sitemapConfig,
      })

      const xml = await get('/sitemap.xml')
      expect(xml).not.toContain('<loc>https://example.com/</loc>')
      expect(xml).toContain('<loc>https://example.com/en/</loc>')
      expect(xml).not.toContain('<loc>https://example.com/fr/</loc>')
      expect(xml).not.toContain('<xhtml:link rel="alternate" hreflang="en" href="https://example.com/"/>')
      expect(xml).toContain('<xhtml:link rel="alternate" hreflang="en" href="https://example.com/en/"/>')
      expect(xml).toContain('<xhtml:link rel="alternate" hreflang="fr" href="https://example.com/fr/"/>')
      expect(xml).not.toContain('<xhtml:link rel="alternate" hreflang="x-default" href="https://example.com/"/>')
    })

    test('strategy "prefix_except_default"', async () => {
      nuxt = await startServer({
        ...config,
        modules,
        i18n: {
          ...nuxtI18nConfig,
          strategy: 'prefix_except_default',
        },
        sitemap: sitemapConfig,
      })

      const xml = await get('/sitemap.xml')
      expect(xml).toContain('<loc>https://example.com/</loc>')
      expect(xml).not.toContain('<loc>https://example.com/en/</loc>')
      expect(xml).not.toContain('<loc>https://example.com/fr/</loc>')
      expect(xml).toContain('<xhtml:link rel="alternate" hreflang="en" href="https://example.com/"/>')
      expect(xml).not.toContain('<xhtml:link rel="alternate" hreflang="en" href="https://example.com/en/"/>')
      expect(xml).toContain('<xhtml:link rel="alternate" hreflang="fr" href="https://example.com/fr/"/>')
      expect(xml).not.toContain('<xhtml:link rel="alternate" hreflang="x-default" href="https://example.com/"/>')
    })

    test('strategy "prefix_and_default"', async () => {
      nuxt = await startServer({
        ...config,
        modules,
        i18n: {
          ...nuxtI18nConfig,
          strategy: 'prefix_and_default',
        },
        sitemap: {
          ...sitemapConfig,
          i18n: {
            defaultLocale: 'x-default',
          },
        },
      })

      const xml = await get('/sitemap.xml')
      expect(xml).toContain('<loc>https://example.com/</loc>')
      expect(xml).not.toContain('<loc>https://example.com/en/</loc>')
      expect(xml).not.toContain('<loc>https://example.com/fr/</loc>')
      expect(xml).not.toContain('<xhtml:link rel="alternate" hreflang="en" href="https://example.com/"/>')
      expect(xml).toContain('<xhtml:link rel="alternate" hreflang="en" href="https://example.com/en/"/>')
      expect(xml).toContain('<xhtml:link rel="alternate" hreflang="fr" href="https://example.com/fr/"/>')
      expect(xml).toContain('<xhtml:link rel="alternate" hreflang="x-default" href="https://example.com/"/>')
    })

    test('locales with iso values', async () => {
      const locales = [
        { code: 'en', iso: 'en-US' },
        { code: 'gb', iso: 'en-GB' },
      ]
      nuxt = await startServer({
        ...config,
        modules,
        i18n: {
          ...nuxtI18nConfig,
          locales,
        },
        sitemap: {
          ...sitemapConfig,
          i18n: {
            defaultLocale: 'en',
            locales,
          },
        },
      })

      const xml = await get('/sitemap.xml')
      expect(xml).toContain('<loc>https://example.com/</loc>')
      expect(xml).toContain('<xhtml:link rel="alternate" hreflang="en-US" href="https://example.com/"/>')
      expect(xml).toContain('<xhtml:link rel="alternate" hreflang="en-GB" href="https://example.com/gb/"/>')
    })
  })

  describe('external options', () => {
    test('default hostname from build.publicPath', async () => {
      nuxt = await startServer({
        ...config,
        build: {
          publicPath: 'https://example.com',
        },
      })

      xml = await get('/sitemap.xml')
      expect(xml).toContain('<loc>https://example.com/</loc>')
    })

    test('default routes from generate.routes', async () => {
      nuxt = await startServer({
        ...config,
        generate: {
          routes: ['test'],
        },
        sitemap: {
          hostname: 'https://example.com/',
        },
      })

      xml = await get('/sitemap.xml')
      expect(xml).toContain('<loc>https://example.com/test</loc>')
    })

    test('custom base from router.base', async () => {
      nuxt = await startServer({
        ...config,
        router: {
          base: '/base',
        },
        sitemap: {
          hostname: 'https://example.com/',
        },
      })

      xml = await get('/base/sitemap.xml')
      expect(xml).toMatchSnapshot()
    })
  })
})

describe('sitemap - multiple configuration', () => {
  let nuxt = null

  beforeAll(async () => {
    nuxt = await startServer({
      ...config,
      sitemap: [
        {
          path: 'sitemap-foo.xml',
          hostname: 'https://example.com/',
        },
        {
          path: 'sitemap-bar.xml',
          hostname: 'https://example.org/',
        },
      ],
    })
  })

  test('sitemap-foo.xml', async () => {
    const xml = await get('/sitemap-foo.xml')
    expect(xml).toMatchSnapshot()
  })

  test('sitemap-bar.xml', async () => {
    const xml = await get('/sitemap-bar.xml')
    expect(xml).toMatchSnapshot()
  })

  afterAll(async () => {
    await nuxt.close()
  })
})

describe('sitemapindex - minimal configuration', () => {
  let nuxt = null

  beforeAll(async () => {
    nuxt = await startServer({
      ...config,
      sitemap: {
        hostname: 'https://example.com/',
        sitemaps: [
          {
            path: '/sitemap-foo.xml',
            routes: ['foo/1', 'foo/2'],
          },
          {
            path: '/sitemap-bar.xml',
            routes: ['bar/1', 'bar/2'],
          },
        ],
      },
    })
  })

  test('sitemapindex.xml', async () => {
    const xml = await get('/sitemapindex.xml')
    expect(xml).toContain('<loc>https://example.com/sitemap-foo.xml</loc>')
    expect(xml).toContain('<loc>https://example.com/sitemap-bar.xml</loc>')
  })

  test('sitemap-foo.xml', async () => {
    const xml = await get('/sitemap-foo.xml')
    expect(xml).toContain('<loc>https://example.com/foo/1</loc>')
    expect(xml).toContain('<loc>https://example.com/foo/2</loc>')
  })

  test('sitemap-bar.xml', async () => {
    const xml = await get('/sitemap-bar.xml')
    expect(xml).toContain('<loc>https://example.com/bar/1</loc>')
    expect(xml).toContain('<loc>https://example.com/bar/2</loc>')
  })

  afterAll(async () => {
    await nuxt.close()
  })
})

describe('sitemapindex - advanced configuration', () => {
  let nuxt = null
  let xml = null
  const today = new Date().toISOString()
  const yesterday = new Date(new Date() - 1000 * 60 * 60 * 24).toISOString()

  beforeAll(async () => {
    nuxt = await startServer({
      ...config,
      sitemap: {
        path: '/sitemapindex.xml',
        hostname: 'https://example.com/',
        lastmod: today,
        sitemaps: [
          {
            path: '/sitemap-foo.xml',
            routes: ['foo/1', 'foo/2'],
            lastmod: yesterday,
          },
          {
            hostname: 'https://example.fr/',
            path: '/sitemap-bar.xml',
            routes: ['bar/1', 'bar/2'],
          },
        ],
        gzip: true,
        xmlNs: 'xmlns="https://example.com/schemas/sitemap/0.9"',
        xslUrl: 'sitemapindex.xsl',
      },
    })

    xml = await get('/sitemapindex.xml')
  })

  test('cascading hostname', () => {
    expect(xml).toContain('<loc>https://example.com/sitemap-foo.xml</loc>')
    expect(xml).toContain('<loc>https://example.fr/sitemap-bar.xml</loc>')
  })

  test('custom lastmod', () => {
    expect(xml).toContain(`<lastmod>${today}</lastmod>`)
    expect(xml).toContain(`<lastmod>${yesterday}</lastmod>`)
  })

  test('etag enabled', async () => {
    const requestOptions = {
      simple: false,
      resolveWithFullResponse: true,
    }

    // 1st call
    let response = await get('/sitemapindex.xml', requestOptions)
    expect(response.statusCode).toEqual(200)
    expect(response.headers.etag).not.toBeUndefined()
    // 2nd call
    response = await get('/sitemapindex.xml', {
      headers: {
        'If-None-Match': response.headers.etag,
      },
      ...requestOptions,
    })
    expect(response.statusCode).toEqual(304)

    // 1st call
    response = await get('/sitemapindex.xml.gz', requestOptions)
    expect(response.statusCode).toEqual(200)
    expect(response.headers.etag).not.toBeUndefined()
    // 2nd call
    response = await get('/sitemapindex.xml.gz', {
      headers: {
        'If-None-Match': response.headers.etag,
      },
      ...requestOptions,
    })
    expect(response.statusCode).toEqual(304)
  })

  test('gzip enabled', async () => {
    const gz = await getGzip('/sitemapindex.xml.gz')
    const sitemap = gunzipSync(gz).toString()
    expect(xml).toEqual(sitemap)
  })

  test('custom XML namespaces', () => {
    expect(xml).toContain('<sitemapindex xmlns="https://example.com/schemas/sitemap/0.9">')
  })

  test('custom XSL', () => {
    expect(xml).toContain('<?xml-stylesheet type="text/xsl" href="sitemapindex.xsl"?>')
  })

  afterAll(async () => {
    await nuxt.close()
  })
})

describe('sitemapindex - custom router base', () => {
  let nuxt = null

  beforeAll(async () => {
    nuxt = await startServer({
      ...config,
      router: {
        base: '/base',
      },
      sitemap: {
        hostname: 'https://example.com/',
        sitemaps: [
          {
            path: '/sitemap-foo.xml',
            routes: ['foo/1', 'foo/2'],
          },
          {
            hostname: 'https://example.fr/',
            path: '/sitemap-bar.xml',
            routes: ['bar/1', 'bar/2'],
          },
        ],
      },
    })
  })

  test('sitemapindex.xml', async () => {
    const xml = await get('/base/sitemapindex.xml')
    expect(xml).toContain('<loc>https://example.com/base/sitemap-foo.xml</loc>')
    expect(xml).toContain('<loc>https://example.fr/base/sitemap-bar.xml</loc>')
  })

  test('sitemap-foo.xml', async () => {
    const xml = await get('/base/sitemap-foo.xml')
    expect(xml).toContain('<loc>https://example.com/base/foo/1</loc>')
    expect(xml).toContain('<loc>https://example.com/base/foo/2</loc>')
  })

  test('sitemap-bar.xml', async () => {
    const xml = await get('/base/sitemap-bar.xml')
    expect(xml).toContain('<loc>https://example.fr/base/bar/1</loc>')
    expect(xml).toContain('<loc>https://example.fr/base/bar/2</loc>')
  })

  afterAll(async () => {
    await nuxt.close()
  })
})

// TODO: describe('sitemapindex - multiple configuration', () => { ... })

describe('sitemap - generate mode', () => {
  test('sitemap.xml', async () => {
    await runGenerate({
      ...config,
      sitemap: {
        hostname: 'https://example.com/',
        exclude: ['/exclude'],
      },
    })

    const xml = readFileSync(resolve(__dirname, '../dist/sitemap.xml'), 'utf8')
    expect(xml).toMatchSnapshot()
  })

  test('sitemap.xml.gz', async () => {
    await runGenerate({
      ...config,
      sitemap: {
        hostname: 'https://example.com/',
        gzip: true,
      },
    })

    const xml = readFileSync(resolve(__dirname, '../dist/sitemap.xml'), 'utf8')
    const gz = readFileSync(resolve(__dirname, '../dist/sitemap.xml.gz'))
    const sitemap = gunzipSync(gz).toString()
    expect(xml).toEqual(sitemap)
  })
})

describe('sitemapindex - generate mode', () => {
  beforeAll(async () => {
    await runGenerate({
      ...config,
      sitemap: {
        hostname: 'https://example.com/',
        sitemaps: [
          {
            path: '/sitemap-foo.xml',
            routes: ['foo/1', 'foo/2'],
          },
          {
            hostname: 'https://example.fr/',
            path: '/sitemap-bar.xml',
            routes: ['bar/1', 'bar/2'],
          },
        ],
        gzip: true,
      },
    })
  })

  test('sitemapindex.xml', () => {
    const xml = readFileSync(resolve(__dirname, '../dist/sitemapindex.xml'), 'utf8')
    expect(xml).toMatchSnapshot()
  })

  test('sitemapindex.xml.gz', () => {
    const xml = readFileSync(resolve(__dirname, '../dist/sitemapindex.xml'), 'utf8')
    const gz = readFileSync(resolve(__dirname, '../dist/sitemapindex.xml.gz'))
    const sitemapindex = gunzipSync(gz).toString()
    expect(xml).toEqual(sitemapindex)
  })

  test('sitemap-foo.xml', () => {
    const xml = readFileSync(resolve(__dirname, '../dist/sitemap-foo.xml'), 'utf8')
    expect(xml).toMatchSnapshot()
  })

  test('sitemap-bar.xml', () => {
    const xml = readFileSync(resolve(__dirname, '../dist/sitemap-bar.xml'), 'utf8')
    expect(xml).toMatchSnapshot()
  })
})
