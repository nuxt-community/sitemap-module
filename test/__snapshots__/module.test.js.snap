// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sitemap - advanced configuration external options custom base from router.base 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:news=\\"http://www.google.com/schemas/sitemap-news/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\" xmlns:mobile=\\"http://www.google.com/schemas/sitemap-mobile/1.0\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>https://example.com/base/child</loc></url><url><loc>https://example.com/base/exclude</loc></url><url><loc>https://example.com/base/filtered</loc></url><url><loc>https://example.com/base/parent/child/subchild</loc></url><url><loc>https://example.com/base/parent/child</loc></url><url><loc>https://example.com/base/parent</loc></url><url><loc>https://example.com/base/sub</loc></url><url><loc>https://example.com/base/sub/sub</loc></url><url><loc>https://example.com/base/</loc></url></urlset>"`;

exports[`sitemap - generate mode sitemap.xml 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:news=\\"http://www.google.com/schemas/sitemap-news/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\" xmlns:mobile=\\"http://www.google.com/schemas/sitemap-mobile/1.0\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>https://example.com/child</loc></url><url><loc>https://example.com/filtered</loc></url><url><loc>https://example.com/parent/child/subchild</loc></url><url><loc>https://example.com/parent/child</loc></url><url><loc>https://example.com/parent</loc></url><url><loc>https://example.com/sub</loc></url><url><loc>https://example.com/sub/sub</loc></url><url><loc>https://example.com/</loc></url></urlset>"`;

exports[`sitemap - minimal configuration sitemap.xml 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:news=\\"http://www.google.com/schemas/sitemap-news/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\" xmlns:mobile=\\"http://www.google.com/schemas/sitemap-mobile/1.0\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>https://example.com/child</loc></url><url><loc>https://example.com/exclude</loc></url><url><loc>https://example.com/filtered</loc></url><url><loc>https://example.com/parent/child/subchild</loc></url><url><loc>https://example.com/parent/child</loc></url><url><loc>https://example.com/parent</loc></url><url><loc>https://example.com/sub</loc></url><url><loc>https://example.com/sub/sub</loc></url><url><loc>https://example.com/</loc></url></urlset>"`;

exports[`sitemap - multiple configuration sitemap-bar.xml 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:news=\\"http://www.google.com/schemas/sitemap-news/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\" xmlns:mobile=\\"http://www.google.com/schemas/sitemap-mobile/1.0\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>https://example.org/child</loc></url><url><loc>https://example.org/exclude</loc></url><url><loc>https://example.org/filtered</loc></url><url><loc>https://example.org/parent/child/subchild</loc></url><url><loc>https://example.org/parent/child</loc></url><url><loc>https://example.org/parent</loc></url><url><loc>https://example.org/sub</loc></url><url><loc>https://example.org/sub/sub</loc></url><url><loc>https://example.org/</loc></url></urlset>"`;

exports[`sitemap - multiple configuration sitemap-foo.xml 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:news=\\"http://www.google.com/schemas/sitemap-news/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\" xmlns:mobile=\\"http://www.google.com/schemas/sitemap-mobile/1.0\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>https://example.com/child</loc></url><url><loc>https://example.com/exclude</loc></url><url><loc>https://example.com/filtered</loc></url><url><loc>https://example.com/parent/child/subchild</loc></url><url><loc>https://example.com/parent/child</loc></url><url><loc>https://example.com/parent</loc></url><url><loc>https://example.com/sub</loc></url><url><loc>https://example.com/sub/sub</loc></url><url><loc>https://example.com/</loc></url></urlset>"`;

exports[`sitemapindex - generate mode sitemap-bar.xml 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:news=\\"http://www.google.com/schemas/sitemap-news/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\" xmlns:mobile=\\"http://www.google.com/schemas/sitemap-mobile/1.0\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>https://example.fr/bar/1</loc></url><url><loc>https://example.fr/bar/2</loc></url><url><loc>https://example.fr/child</loc></url><url><loc>https://example.fr/exclude</loc></url><url><loc>https://example.fr/filtered</loc></url><url><loc>https://example.fr/parent/child/subchild</loc></url><url><loc>https://example.fr/parent/child</loc></url><url><loc>https://example.fr/parent</loc></url><url><loc>https://example.fr/sub</loc></url><url><loc>https://example.fr/sub/sub</loc></url><url><loc>https://example.fr/</loc></url></urlset>"`;

exports[`sitemapindex - generate mode sitemap-foo.xml 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:news=\\"http://www.google.com/schemas/sitemap-news/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\" xmlns:mobile=\\"http://www.google.com/schemas/sitemap-mobile/1.0\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>https://example.com/foo/1</loc></url><url><loc>https://example.com/foo/2</loc></url><url><loc>https://example.com/child</loc></url><url><loc>https://example.com/exclude</loc></url><url><loc>https://example.com/filtered</loc></url><url><loc>https://example.com/parent/child/subchild</loc></url><url><loc>https://example.com/parent/child</loc></url><url><loc>https://example.com/parent</loc></url><url><loc>https://example.com/sub</loc></url><url><loc>https://example.com/sub/sub</loc></url><url><loc>https://example.com/</loc></url></urlset>"`;

exports[`sitemapindex - generate mode sitemapindex.xml 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><sitemapindex xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\"><sitemap><loc>https://example.com/sitemap-foo.xml</loc></sitemap><sitemap><loc>https://example.fr/sitemap-bar.xml</loc></sitemap></sitemapindex>"`;
